sdk/eventhub/event-hubs/src/eventHubBufferedProducerClient.ts

```diff
@@ -0,0 +1,315 @@
+   * Publishing will take place at a nondeterministic point in the future as the buffer is processed.
+   *
+   * @param events - An array of {@link EventData} or `AmqpAnnotatedMessage`.
+   * @param options - A set of options that can be specified to influence the way in which
+   * events are sent to the associated Event Hub.
+   * - `abortSignal`  : A signal the request to cancel the send operation.
+   * - `partitionId`  : The partition this set of events will be sent to. If set, `partitionKey` can not be set.
+   * - `partitionKey` : A value that is hashed to produce a partition assignment. If set, `partitionId` can not be set.
+   * @returns The total number of events that are currently buffered and waiting to be published, across all partitions.
+   */
+  async enqueueEvents(
+    events: EventData[] | AmqpAnnotatedMessage[],
+    options: EnqueueEventOptions = {}
+  ): Promise<number> {
+    throw new Error(`Not implemented ${events}, ${options}`);
+  }
+
+  /**
+   * Attempts to publish all events in the buffer immediately.
+   * This may result in multiple batches being published,
+   * the outcome of each of which will be individually reported by
+   * the `onSendEventsSuccessHandler` and `onSendEventsErrorHandler` handlers.
+   *
+   * @param options - The set of options to apply to the operation call.
+   */
+  async flush(options: OperationOptions = {}): Promise<void> {
+    throw new Error(`Not implemented ${options}`);
+  }
+
+  /**
+   * Provides the Event Hub runtime information.
+   * @param options - The set of options to apply to the operation call.
+   * @returns A promise that resolves with information about the Event Hub instance.
+   * @throws Error if the underlying connection has been closed, create a new EventHubBufferedProducerClient.
+   * @throws AbortError if the operation is cancelled via the abortSignal.
+   */
+  getEventHubProperties(options: GetEventHubPropertiesOptions = {}): Promise<EventHubProperties> {
+    return this._context.managementSession!.getEventHubProperties({
+      ...options,
+      retryOptions: this._clientOptions.retryOptions
+    });
+  }
+
+  /**
+   * Provides the id for each partition associated with the Event Hub.
+   * @param options - The set of options to apply to the operation call.
+   * @returns A promise that resolves with an Array of strings representing the id for
+   * each partition associated with the Event Hub.
+   * @throws Error if the underlying connection has been closed, create a new EventHubBufferedProducerClient.
+   * @throws AbortError if the operation is cancelled via the abortSignal.
+   */
+  getPartitionIds(options: GetPartitionIdsOptions = {}): Promise<Array<string>> {
+    return this._context
+      .managementSession!.getEventHubProperties({
+        ...options,
+        retryOptions: this._clientOptions.retryOptions
+      })
+      .then((eventHubProperties) => {
+        return eventHubProperties.partitionIds;
+      });
```

Comment:
NIT:
```suggestion
      .then((eventHubProperties) => eventHubProperties.partitionIds);
```

Reply:
This will change in the implementation to just return `getPartitionIds` on a local EventHubProducerClient, so I'll defer for now.

Reply:
@chradek I do not mind deferring, it is a small nit but I am curious what did you mean? if the two statements do not behave the same, what is the difference?